src/store.c

```diff
@@ -68,42 +71,82 @@ struct rill_store
 
     struct header *head;
     struct vals *vals;
-    uint8_t *data;
-    struct index *index;
+
+    uint8_t *data_a;
+    uint8_t *data_b;
+    struct index *index_a;
+    struct index *index_b;
     uint8_t *end;
 };
 
+struct rill_space
+{
+    size_t header_bytes;
+    size_t index_bytes[2];
+    size_t pairs_bytes[2];
+};
```

Comment:
I would just put this in the interface directly as there's no internal implementation details that needs to be hidden. It's going to be alot easier for the user to just access the struct directly and be able to store that struct on the stack with no need to clean it up.